pkg/test/framework/components/istio/operator.go

```diff
@@ -207,6 +207,18 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 			})
 		}
 	}
+
+	// patch istiod deployment with ISTIOD_CUSTOM_HOST
+	if isCentralIstio(env, cfg) {
```

Comment:
Can we just patch the yaml before we deploy it in the first place?

Reply:
You can use --set for the ISTIOD_CUSTOM_HOST env var - but that only works if you know the IP of ingress gateway prior to deploy. There is an issue for making this item configurable: istio/api#1414